File: spec/python/test_eof_exception_u4.py
from base_test import BaseTest
from eof_exception_u4 import EofExceptionU4

class TestEofExceptionU4(BaseTest):
    def test_eof_exception_u4(self):
        with self.assertRaisesRegex(EOFError, "^requested \d+ bytes, but only \d+ bytes available$"):
            with EofExceptionU4.from_file('src/term_strz.bin') as r:
                pass

File: spec/lua/test_eof_exception_u4.lua
local luaunit = require("luaunit")

require("eof_exception_u4")

TestEofExceptionU4 = {}

function TestEofExceptionU4:test_eof_exception_u4()
    luaunit.assertErrorMsgMatches(".+: requested %d+ bytes, but got only %d+ bytes", EofExceptionU4.from_file, EofExceptionU4, "src/term_strz.bin")
end
